From b8d07281b1577a2278c0f96baffbcff39ab308bd Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Tue, 6 Apr 2021 11:31:19 -0400
Subject: [PATCH 28/36] Make AuroaServices conditional

Make it so that it builds in only when USE_FOSS blag is enabled.
Change-Id: If2834253841e30eee6cb7b84b666fa9543cda255
---
 foss_privapp/Android.mk                       |  44 ++++++++++++++++++
 .../AuroraServices_v1.1.1.apk                 | Bin
 2 files changed, 44 insertions(+)
 create mode 100644 foss_privapp/Android.mk
 rename {privapp => foss_privapp}/AuroraServices_v1.1.1.apk (100%)

diff --git a/foss_privapp/Android.mk b/foss_privapp/Android.mk
new file mode 100644
index 0000000..22bd39b
--- /dev/null
+++ b/foss_privapp/Android.mk
@@ -0,0 +1,44 @@
+#
+# 2021 Bliss Roms - Adapted from Android-x86 Project
+# Original Copyright (C) 2011-2015 The Android-x86 Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+
+LOCAL_PATH := $(call my-dir)
+ifneq ("$(wildcard vendor/foss/bin/*)","")
+
+LOCAL_APPS := $(subst $(LOCAL_PATH)/,,$(wildcard $(LOCAL_PATH)/*$(COMMON_ANDROID_PACKAGE_SUFFIX)))
+
+define include-app
+include $$(CLEAR_VARS)
+
+LOCAL_LIBS := $$(shell zipinfo -1 $$(LOCAL_PATH)/$(1) | grep ^lib/ | grep -v /$$$$)
+
+LOCAL_MODULE := $$(basename $(1))
+LOCAL_MODULE_TAGS := optional
+LOCAL_MODULE_CLASS := APPS
+LOCAL_MODULE_SUFFIX := $$(suffix $(1))
+LOCAL_BUILT_MODULE_STEM := package.apk
+LOCAL_CERTIFICATE := PRESIGNED
+LOCAL_SRC_FILES := $(1)
+LOCAL_PRIVILEGED_MODULE := true
+LOCAL_PRIVATE_PLATFORM_APIS := true 
+LOCAL_DEX_PREOPT := false
+LOCAL_MODULE_TARGET_ARCH := $$(call get-prebuilt-src-arch,$$(notdir $$(patsubst %/,%,$$(dir $$(LOCAL_LIBS)))))
+LOCAL_PREBUILT_JNI_LIBS := $$(addprefix @,$$(filter lib/$$(LOCAL_MODULE_TARGET_ARCH)/%,$$(LOCAL_LIBS)))
+#$$(info $$(LOCAL_MODULE) LOCAL_MODULE_TARGET_ARCH=$$(LOCAL_MODULE_TARGET_ARCH))
+#$$(info $$(LOCAL_MODULE) LOCAL_PREBUILT_JNI_LIBS=$$(LOCAL_PREBUILT_JNI_LIBS))
+include $$(BUILD_PREBUILT)
+
+ALL_DEFAULT_INSTALLED_MODULES += $$(LOCAL_INSTALLED_MODULE)
+endef
+
+$(foreach a,$(LOCAL_APPS),$(eval $(call include-app,$(a))))
+else
+include $(call all-subdir-makefiles)
+endif
diff --git a/privapp/AuroraServices_v1.1.1.apk b/foss_privapp/AuroraServices_v1.1.1.apk
similarity index 100%
rename from privapp/AuroraServices_v1.1.1.apk
rename to foss_privapp/AuroraServices_v1.1.1.apk
-- 
2.17.1

